cdeaneGit/islandora_solution_pack_video
# Update derivatives.inc

Add check for 'ffmpeg -version', revise command line arguments as needed.

**JIRA Ticket**: (link)
IslandoraISLANDORA-2062 (https://jira.duraspace.org/browse/ISLANDORA-2062?filter=14301)

# What does this Pull Request do?
This pull request addresses a single problem that is taken care of by adding code to a single file:

Revise islandora_solution_pack_video to detect ffmpeg version and adjust the command line to allow or exclude the arguments ' -strict -2 ' .

The file that changes is:
islandora_solution_pack_video/includes/derivatives.inc

The result should be a command line that executes with no warnings or error messages about Strict Mode. 

# How should this be tested?
1. A small video file should be ingested to a repository using the default version of ffmpeg.
2. Reinstall islandora_solution_pack_video with a different version of ffmpeg by revising the file:
   islandora_solution_pack_video/tests/scripts/ffmpeg-install.sh 
   to use a different target for the wget command.
3. Then repeat step 1 ingest.
4. The ingest should go smoothly for each case.

# To reproduce the problem:
1. Install ffmpeg version 1.x.x with islandora_solution_pack_video
2. Use the version of derivatives.inc that does not change the command line.
3. You should see warnings.



# Additional Notes:
New releases of ffmpeg: https://ffmpeg.org/download.html

Older releases of ffmpeg: https://ffmpeg.org/olddownload.html

Example:
* Does this change require documentation to be updated? No.
* Does this change add any new dependencies? No.
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? No.
* Could this change impact execution of existing code? I hope so.

# Interested parties
Tag (@DonRichards, @Islandora/7-x-1-x-committers)
